openedx/core/djangoapps/credit/models.py

```diff
@@ -508,6 +508,22 @@ def remove_requirement_status(cls, username, requirement):
             log.error(log_msg)
             return
 
+    @classmethod
+    @transaction.atomic
+    def retire_user(cls, retired_username, retiring_username=""):
+        try:
+            requirement_statuses = cls.objects.filter(username=retired_username)
+            for requirement_status in requirement_statuses:
+                requirement_status.username = retiring_username
+                requirement_status.reason = {}
+                requirement_status.save()
+        except cls.DoesNotExist:
+            log.info(
+                u'Can not retire requirement statuses for user "%s" because the user could not be found ',
+                retiring_username
```

Comment:
I think this should really be `retired_username`, right (whatever the original, non-hashed username is)?